avm/ptn/app-service-lza/hosting-environment/modules/front-door/front-door.module.bicep

```diff
@@ -0,0 +1,287 @@
+
+@description('Required. Whether to enable deployment telemetry.')
+param enableTelemetry bool
+
+@description('Optional. Diagnostic Settings for FrontDoor')
+param diagnosticSettings diagnosticSettingFullType[]?
+
+@description('Required. Name of the AFD profile.')
+param afdName string
+
+@description('Name of the endpoint under the profile which is unique globally.')
+param endpointName string
+
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+@description('AFD Endpoint State')
+param endpointEnabled string = 'Enabled'
+
+@description('Optional. Endpoint tags.')
+param tags object = {}
+
+@allowed([
+  'Standard_AzureFrontDoor'
+  'Premium_AzureFrontDoor'
+])
+@description('Required. The pricing tier (defines a CDN provider, feature list and rate) of the CDN profile.')
+param skuName string
+
+// @description('Custom Domain List')
+// param customDomains array
+
+@description('The name of the Origin Group')
+param originGroupName string
+
+@description('Origin List')
+param origins array
+
+@allowed([
+  'Block'
+  'Log'
+  'Redirect'
+])
+param wafRuleSetAction string = 'Block'
+
+@description('optional, default value Enabled. ')
+@allowed([
+  'Enabled'
+  'Disabled'
+])
+param wafPolicyState string = 'Enabled'
+
+@description('optional, default value Prevention. ')
+@allowed([
+  'Detection'
+  'Prevention'
+])
+param wafPolicyMode string = 'Prevention'
+
+// Create an Array of all Endpoint which includes customDomain Id and afdEndpoint Id
+// This array is needed to be attached to Microsoft.Cdn/profiles/securitypolicies
+// var customDomainIds = [for (domain, index) in customDomains: {id: custom_domains[index].id}]
+// var afdEndpointIds = [{id: endpoint.id}]
+// var endPointIdsForWaf = union(customDomainIds, afdEndpointIds)
+//var endPointIdsForWaf = [{ id: endpoint.id }]
+
+// @description('Default Content to compress')
+// var contentTypeCompressionList = [
+//   'application/eot'
+//   'application/font'
+//   'application/font-sfnt'
+//   'application/javascript'
+//   'application/json'
+//   'application/opentype'
+//   'application/otf'
+//   'application/pkcs7-mime'
+//   'application/truetype'
+//   'application/ttf'
+//   'application/vnd.ms-fontobject'
+//   'application/xhtml+xml'
+//   'application/xml'
+//   'application/xml+rss'
+//   'application/x-font-opentype'
+//   'application/x-font-truetype'
+//   'application/x-font-ttf'
+//   'application/x-httpd-cgi'
+//   'application/x-javascript'
+//   'application/x-mpegurl'
+//   'application/x-opentype'
+//   'application/x-otf'
+//   'application/x-perl'
+//   'application/x-ttf'
+//   'font/eot'
+//   'font/ttf'
+//   'font/otf'
+//   'font/opentype'
+//   'image/svg+xml'
+//   'text/css'
+//   'text/csv'
+//   'text/html'
+//   'text/javascript'
+//   'text/js'
+//   'text/plain'
+//   'text/richtext'
+//   'text/tab-separated-values'
+//   'text/xml'
+//   'text/x-script'
+//   'text/x-component'
+//   'text/x-java-source'
+// ]
```

Comment:
Does this need to still be here?

Reply:
My understanding was that the original author wanted to keep it to implement in the next version